src/FsCheck/Runner.fs

```diff
@@ -48,34 +48,74 @@ type ParallelRunConfig =
-      EndSize       : int
-      ///If set, suppresses the output from the test if the test is successful.
-      QuietOnSuccess: bool
-      ///What to print when new arguments args are generated in test n
-      Every         : int -> list<obj> -> string
-      ///What to print every time a counter-example is successfully shrunk
-      EveryShrink   : list<obj> -> string 
-      ///The Arbitrary instances on this class will be merged in back to front order, i.e. instances for the same generated type at the front
-      ///of the list will override those at the back. The instances on Arb.Default are always known, and are at the back (so they can always be
-      ///overridden)
-      Arbitrary     : list<Type>
-      ///A custom test runner, e.g. to integrate with a test framework like xUnit or NUnit. 
-      Runner        : IRunner
-      ///If set, inputs for property generation and property evaluation will be run in parallel. 
-      ParallelRunConfig : ParallelRunConfig option
-    }
+type Config = private Config of
+                {| ///The maximum number of tests that are run.
+                MaxTest       : int
+                ///The maximum number of tests where values are rejected, e.g. as the result of ==>
+                MaxRejected   : int
+                ///If set, the seed to use to start testing. Allows reproduction of previous runs.
+                Replay        : Replay option
+                ///Name of the test.
+                Name          : string
+                ///The size to use for the first test.
+                StartSize     : int
+                ///The size to use for the last test, when all the tests are passing. The size increases linearly between Start- and EndSize.
+                EndSize       : int
+                ///If set, suppresses the output from the test if the test is successful.
+                QuietOnSuccess: bool
+                ///What to print when new arguments args are generated in test n
+                Every         : int -> list<obj> -> string
+                ///What to print every time a counter-example is successfully shrunk
+                EveryShrink   : list<obj> -> string 
+                ///The Arbitrary instances on this class will be merged in back to front order, i.e. instances for the same generated type at the front
+                ///of the list will override those at the back. The instances on Arb.Default are always known, and are at the back (so they can always be
+                ///overridden)
+                Arbitrary     : list<Type>
+                ///A custom test runner, e.g. to integrate with a test framework like xUnit or NUnit. 
+                Runner        : IRunner
+                ///If set, inputs for property generation and property evaluation will be run in parallel. 
+                ParallelRunConfig : ParallelRunConfig option
```

Comment:
Can you move the docs to the `WithX` methods?

Reply:
I've added comments to the WithX methods and moved the comments from the record.

I also noticed that the comments were not being generated for the anonymous record, so I copied the fields (and their comments) into members of Config.